src/frontend/change_buffer.rs

```diff
@@ -93,28 +110,63 @@ fn read_keys(
                 exit.send_default();
                 return Ok(());
             }
-            (IndexSelection::Single(i), KeyCode::Char('j')) => {
-                Some(IndexSelection::Single(usize::min(i + 1, max)))
-            }
-            (IndexSelection::Single(i), KeyCode::Char('k')) => {
-                Some(IndexSelection::Single(i.saturating_sub(1)))
-            }
-            (IndexSelection::Single(i), KeyCode::Char('x')) => {
-                if i != (i + 1).clamp(0, max) {
-                    Some(IndexSelection::Range(i, usize::min(i + 1, max)))
-                } else {
-                    Some(IndexSelection::Single(i))
+
+            (IndexSelection::Single(i), KeyCode::Char(key)) => {
+                match resolve_command(&config, key) {
+                    Some(Command::Static(StaticCommand::MoveVisualLineDown)) => {
+                        Some(IndexSelection::Single(usize::min(i + 1, max)))
+                    }
+
+                    Some(Command::Static(StaticCommand::MoveVisualLineUp)) => {
+                        Some(IndexSelection::Single(usize::min(i.wrapping_sub(1), max)))

```

Comment:
Let's avoid wrapping selections. We'll add goto commands later on via minor modes, a la `gg` and `ge`